### PR TITLE
UGENE-7892 Find interrupted and degenerate sites

### PR DESCRIPTION
**Pull request by DmitriiSukhomlinov**
See [issue comment](https://ugene.dev/tracker/browse/UGENE-7892?focusedCommentId=32243&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-32243) for details.

### CODE REVIEW

**Review by yalgaer — CHANGES_REQUESTED** [review]
Please do not test the algorithm with GUI tests. Use XML (or even API) tests: it is 100x times faster and allows to have a better coverage (have more tests).

File: src/plugins/enzymes/src/FindEnzymesAlgorithm.h [hunk 1]
```diff
@@ -90,7 +89,7 @@ class FindEnzymesAlgorithm {
                 buf.append(dnaseq.mid(startPos));
                 buf.append(dnaseq.mid(0, size));
                 for (int s = 0; s < size; s++) {
-                    bool match = matchSite(buf.constData() + s, pattern, plen, unknownChar, fn);
+                    bool match = matchSite(buf.constData() + s, pattern, plen, fn);
```
Comment (yalgaer) [inline src/plugins/enzymes/src/FindEnzymesAlgorithm.h]: The change makes the algorithm to produce wrong results. Please add more tests.
Before:
image

After:
image

Comment (DmitriiSukhomlinov) [reply src/plugins/enzymes/src/FindEnzymesAlgorithm.h]: Fixed.

**Comment by DmitriiSukhomlinov** [comment]
> Please do not test the algorithm with GUI tests. Use XML (or even API) tests: it is 100x times faster and allows to have a better coverage (have more tests).

OK, replaced with XML tests + upgraded some tests infrastructure (it is possible to set annotation qualifiers now - required for test_0009.xml)

File: src/plugins/enzymes/src/FindEnzymesAlgorithm.h (hunk 1 above)
Comment (DmitriiSukhomlinov) [reply src/plugins/enzymes/src/FindEnzymesAlgorithm.h]: test010.xml

**Comment by DmitriiSukhomlinov** [comment]
@yalgaer please, approve if everything is OK. Windows failed test is some random one, Linux failed because agent has been disconnected, I'm sure no problems expected on re-running

**Review by yalgaer — APPROVED** [review]

**Pull request merged**